### PR TITLE
Prevent dataMessage callback errors from killing a meeting

### PR DESCRIPTION
**Pull request by schontz**
**Issue #:** #2974

**Description of changes:**

Do not create a fatal error when the user-defined callbacks throw an error.

**Testing:**

Unit tested. POC tested in internal application.

**Checklist:**

1. Have you successfully run `npm run build:release` locally?
Yes

2. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved?
No

3. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved?
No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
Yes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,18 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased]
```
Comment (hensmi-amazon) [inline CHANGELOG.md:8]: I know it's an odd choice, but 3.26 is actually the unreleased section :)

File: src/realtimecontroller/DefaultRealtimeController.ts [hunk 2]
```diff
@@ -413,13 +414,12 @@ export default class DefaultRealtimeController implements RealtimeController {
...
-          fn(dataMessage);
-        }
-      }
+      iterateEvery(this.state.receiveDataMessageCallbacks.get(dataMessage.topic), fn => {
+        fn(dataMessage);
+      });
     } catch (e) {
-      this.onError(e);
+      // We don't want to throw to our caller, but we still want to surface the error in the console
+      Promise.reject(e);
```
Comment (ltrung) [inline src/realtimecontroller/DefaultRealtimeController.ts:422]: What is the issue you ran into? I think it is intentional in any real time API here to throw a hard error.

Comment (hensmi-amazon) [reply src/realtimecontroller/DefaultRealtimeController.ts:422]: There was a parsing issue in the app code handling transcription that threw an exception. The issue is that the audio video controller listens to fatals and shuts down the meeting, even though something as simple as a parsing bug really shouldn't be considered fatal

**Comment by hensmi-amazon** [comment]
Moved to https://github.com/aws/amazon-chime-sdk-js/pull/2976

**Pull request closed without merging**